test/plugins/FakeTimeStampDataGenerator.cpp

```diff
@@ -0,0 +1,190 @@
+  : dunedaq::appfwk::DAQModule(name)
+  , thread_(std::bind(&FakeTimeStampDataGenerator::do_work, this, std::placeholders::_1))
+  , outputQueue_()
+  , queueTimeout_(100)
+  , generator()
+{
+  register_command("conf",  &FakeTimeStampDataGenerator::do_configure);
+  register_command("start", &FakeTimeStampDataGenerator::do_start);
+  register_command("stop",  &FakeTimeStampDataGenerator::do_stop);
+  register_command("scrap", &FakeTimeStampDataGenerator::do_scrap);
+}
+
+void
+FakeTimeStampDataGenerator::init(const nlohmann::json& init_data)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering init() method";
+  auto ini = init_data.get<appfwk::app::ModInit>();
+  for (const auto& qi : ini.qinfos) {
+    if (qi.dir != "output") {
+      continue;                 // skip all but "output" direction
+    }
+    try
+    {
+      //outputQueue_.emplace_back(new sink_t(qi.inst));
+      outputQueue_.reset(new sink_t(qi.inst));
+    }
+    catch (const ers::Issue& excpt)
+    {
+      throw dunedaq::dunetrigger::InvalidQueueFatalError(ERS_HERE, get_name(), qi.name, excpt);
+    }
+  }
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting init() method";
+}
+
+void
+FakeTimeStampDataGenerator::get_info(opmonlib::InfoCollector& /*ci*/, int /*level*/)
+{
+}
+
+void
+FakeTimeStampDataGenerator::do_configure(const nlohmann::json& /*obj*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_configure() method";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_configure() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_start(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_start() method";
+  thread_.start_working_thread();
+  TLOG() << get_name() << " successfully started";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_start() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_stop(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_stop() method";
+  thread_.stop_working_thread();
+  TLOG() << get_name() << " successfully stopped";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_stop() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_scrap(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_scrap() method";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_scrap() method";
+}
+
+ std::vector<triggeralgs::TimeStampedData>
+FakeTimeStampDataGenerator::GetTimestamp()
+{
+  std::vector<triggeralgs::TimeStampedData> tsds;
+  triggeralgs::TimeStampedData tsd{};
+
+  int signaltype = rdm_signaltype(generator);
+
+  auto tsd_start_time = std::chrono::steady_clock::now();
+  tsd.time_stamp = (uint64_t)pd_clock(tsd_start_time.time_since_epoch()).count();
+  tsd.signal_type = signaltype;
+  auto now = std::chrono::steady_clock::now();
```

Comment:
The counter is just meant to increase by one for each TSD produced, not hold a timestamp.